tpchgen/src/dates.rs

```diff
@@ -139,8 +127,8 @@ const fn make_date(index: i32) -> NaiveDate {
     let dy =
         d - MONTH_YEAR_DAY_START[(m - 1) as usize] - if is_leap_year(y) && m > 2 { 1 } else { 0 };
 
-    // Create date from year, month, day
-    NaiveDate::from_ymd_opt(1900 + y, m as u32, dy as u32).unwrap()
+    // Create a date from year, month, day
+    format!("{:04}-{:02}-{:02}", 1900 + y, m, dy)
```

Comment:
I think the conversion to y,m,d without String conversion will be super helpful for others (like @scsmithr) who are using this crate programatically. I'll make a PR to expose just this logic as well as a follow on